File: python-example/scripts/uninstall.py
#!/usr/bin/env python3
import sys

import vdi

#
# Input Handling
#

if len(sys.argv) < 2:
    vdi.log_message("called with too few arguments, callers must provide a VDI dataset ID")
    sys.exit(vdi.EXIT_CODE_UNEXPECTED_ERROR)

VDI_ID = sys.argv[1]

vdi.verify_vdi_id(VDI_ID)

DB_HOST = vdi.require_env("DB_HOST")
DB_PORT = vdi.require_env("DB_PORT")
DB_NAME = vdi.require_env("DB_NAME")
DB_USER = vdi.require_env("DB_USER")
DB_PASS = vdi.require_env("DB_PASS")

#
# Script Body
#

# TODO: uninstall the dataset!

File: go-example/scripts/bin/uninstall.go
package main

import (
	"handlers/pkg/vdi"
	"os"
)

//
// Parameter Handling
//

func main() {
	if len(os.Args) < 2 {
		vdi.LogMessage("called with too few arguments, callers must provide a VDI dataset ID")
		os.Exit(vdi.EXIT_CODE_UNEXPECTED_ERROR)
	}

	vdiID := os.Args[1]

	vdi.VerifyVDIID(vdiID)

	env := map[string]string{
		"DB_HOST": vdi.RequireEnv("DB_HOST"),
		"DB_PORT": vdi.RequireEnv("DB_PORT"),
		"DB_NAME": vdi.RequireEnv("DB_NAME"),
		"DB_USER": vdi.RequireEnv("DB_USER"),
		"DB_PASS": vdi.RequireEnv("DB_PASS"),
	}

	runUninstall(vdiID, env)
}

//
// Script Body
//

func runUninstall(vdiID string, env map[string]string) {

}
